src/basic/log.c

```diff
@@ -588,6 +592,21 @@ static int log_do_header(
         return 0;
 }
 
+static void log_do_context(struct iovec *iovec, size_t iovec_len, size_t *n) {
+        assert(iovec);
+        assert(iovec_len >= 2);
+        assert(n);
+
+        LIST_FOREACH(ll, c, _log_context)
+                STRV_FOREACH(s, c->fields) {
+                        if (*n >= iovec_len - 2)
```

Comment:
maybe write as `if (*n + 2 >= iovec_len)` to avoid the underflow and the assert check?